modules/tinymce/src/plugins/table/main/ts/selection/SelectionTargets.ts

```diff
@@ -134,7 +155,17 @@ export const getSelectionTargets = (editor: Editor, selections: Selections): Sel
     onSetup(api, (targets) => isCaption(targets.element) || getClipboardData().isNone() || isDisabledFromLocked(lockedDisable));
   const onSetupMergeable = (api: UiApi) => onSetup(api, (_targets) => isDisabledForSelection('mergeable'));
   const onSetupUnmergeable = (api: UiApi) => onSetup(api, (_targets) => isDisabledForSelection('unmergeable'));
-
+  const onSetupTableWithCaption = (api: Toolbar.ToolbarToggleButtonInstanceApi) => {
```

Comment:
This needs to work for both menu items and toolbar buttons, so please create a new `UiToggleApi` and use that where relevant. eg:
```
type UiApi = Menu.MenuItemInstanceApi | Toolbar.ToolbarButtonInstanceApi;
type UiToggleApi = Menu.ToggleMenuItemInstanceApi | Toolbar.ToolbarToggleButtonInstanceApi;
```

- [ ] Fix incorrect types